webapp-django/crashstats/crashstats/views.py

```diff
@@ -817,24 +893,293 @@ def append_row_blob(blob, labels):
         assert len(row) == len(head_row), (len(row), len(head_row))
         writer.writerow(row)
 
-    # add the totals
-    totals_labels = (
-        ('crashes', 'Crashes'),
-        ('adu', 'ADI'),
-        ('throttle', 'Throttle'),
-        ('ratio', 'Ratio'),
-    )
-    row = ['Total']
+    return response
 
-    for version in versions:
-        product_version = '%s:%s' % (product, version)
+
+@pass_default_context
+def crashes_per_user(request, default_context=None):
+    context = default_context or {}
+
+    context['products'] = context['currentproducts']['products']
+
+    platforms_api = models.Platforms()
+    platforms = platforms_api.get()
+    form_class = forms.DailyFormByVersion
+
+    date_range_types = ['report', 'build']
+    hang_types = ['any', 'crash', 'hang-p']
+
+    form = form_class(
+        context['currentversions'],
+        platforms,
+        data=request.GET,
+        date_range_types=date_range_types,
+        hang_types=hang_types,
+    )
+    if form.is_valid():
+        params = form.cleaned_data
+        params['product'] = params.pop('p')
+        params['versions'] = sorted(list(set(params.pop('v'))), reverse=True)
         try:
-            blob = data['totals'][product_version]
+            params['platforms'] = params.pop('os')
         except KeyError:
-            continue
-        append_row_blob(blob, totals_labels)
-    writer.writerow(row)
-    return response
+            params['platforms'] = None
+    else:
```

Comment:
Why not reverse this ``if`` and gain one level of indentation, since you have a ``return`` here?